recipes/default.rb

```diff
@@ -20,11 +20,25 @@
   ntp
   qemu-user-static
   sudo
+  x11-xserver-utils
   wget
 ].each do |pkg|
   package pkg
 end
 
+cookbook_file '/etc/X11/xorg.conf' do
+  source 'xorg.conf.no_gpu'
+  mode "0744"
+  not_if "ls /dev/nvidia*"
+end
+cookbook_file '/etc/X11/xorg.conf' do
+  source 'xorg.conf.nvidia'
+  mode "0744"
+  only_if "ls /dev/nvidia*"
+end
```

Comment:
This is an interesting pattern for doing conditional file sources. I'm not sure if I like it yet. I may post an RFC to see what's usually done here. What I probably would have done is set an attribute based on the presence or absence of an nvidia card and then used an `if node['nvidia_device']` conditional to use one source or the other.

I don't think this needs to change unless you like that approach better.

Reply:
I like the way of expressing conditions using the `not_if` or `only_if`. Nothing solid to defend that instead of the `if` approach, just personal opinion. I'll leave it this way by now.

Reply:
Is it because of linting reasons the `only_if` line is last?

For readability having it as the first line would be almost a compromise between what @nuclearsandwich describes and the current approach.